include/exadg/compressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -181,70 +214,87 @@ PostProcessor<dim, Number>::initialize_additional_vectors()
+
+      vorticity.type              = SolutionFieldType::vector;
+      vorticity.name              = "vorticity";
+      vorticity.dof_handler       = &navier_stokes_operator->get_dof_handler_vector();
+      vorticity.initialize_vector = [&](std::shared_ptr<VectorType> & dst) {
+        dst = std::make_shared<VectorType>();
+        navier_stokes_operator->initialize_dof_vector_dim_components(*dst);
+      };
+      vorticity.recompute_solution_field = [&](VectorType & dst, VectorType const & src) {
+        navier_stokes_operator->compute_vorticity(dst, src);
+      };
+
+      vorticity.reinit();
+    }
+
+    // divergence
+    if(pp_data.output_data.write_divergence)
+    {
+      AssertThrow(pp_data.output_data.write_velocity == true,
+                  dealii::ExcMessage("You need to activate write_velocity."));
+
+      divergence.type              = SolutionFieldType::scalar;
+      divergence.name              = "velocity_divergence";
+      divergence.dof_handler       = &navier_stokes_operator->get_dof_handler_scalar();
+      divergence.initialize_vector = [&](std::shared_ptr<VectorType> & dst) {
+        dst = std::make_shared<VectorType>();
+        navier_stokes_operator->initialize_dof_vector_scalar(*dst);
+      };
+      divergence.recompute_solution_field = [&](VectorType & dst, VectorType const & src) {
+        navier_stokes_operator->compute_divergence(dst, src);
+      };
+
+      divergence.reinit();
+    }
   }
 
   // temperature
   if(pp_data.output_data.write_temperature)
   {
-    temperature.type        = SolutionFieldType::scalar;
-    temperature.name        = "temperature";
-    temperature.dof_handler = &navier_stokes_operator->get_dof_handler_scalar();
-    navier_stokes_operator->initialize_dof_vector_scalar(temperature.get_vector_reference());
+    temperature.type              = SolutionFieldType::scalar;
+    temperature.name              = "temperature";
+    temperature.dof_handler       = &navier_stokes_operator->get_dof_handler_scalar();
+    temperature.initialize_vector = [&](std::shared_ptr<VectorType> & dst) {
+      dst = std::make_shared<VectorType>();
+      navier_stokes_operator->initialize_dof_vector_scalar(*dst);
+    };
     temperature.recompute_solution_field = [&](VectorType & dst, VectorType const & src) {
       navier_stokes_operator->compute_temperature(dst, src);
     };
 
-    additional_fields_vtu.push_back(&temperature);
-  }
-
-  // vorticity
-  if(pp_data.output_data.write_vorticity)
-  {
-    vorticity.type        = SolutionFieldType::vector;
-    vorticity.name        = "vorticity";
-    vorticity.dof_handler = &navier_stokes_operator->get_dof_handler_vector();
-    navier_stokes_operator->initialize_dof_vector_dim_components(vorticity.get_vector_reference());
-    vorticity.recompute_solution_field = [&](VectorType & dst, VectorType const &) {
-      navier_stokes_operator->compute_vorticity(dst, velocity.get_vector());
```

Comment:
This should be avoided. If I see it correctly, the code enforces that `velocity` is set up if `write_vorticity`, but it does not enforce that the `velocity` vector is up-to-date.